### PR TITLE
add tenant support for debugger server

### PR DESCRIPTION
**Pull request by Sathiyakugan**

### CODE REVIEW

File: modules/server-endpoints/language-server/src/main/java/org/wso2/carbon/identity/developer/lsp/debug/runtime/DebugSessionManagerImpl.java [hunk 1]
```diff
@@ -86,13 +88,29 @@ public void destroy() {
     @Override
     public void addSession(Session session) {
 
-        activeDebugSessions.put(session, createSession(session));
+        if (getTenantDomain(session) != null && !activeDebugSessions.containsKey(getTenantDomain(session))) {
+            activeDebugSessions.put(getTenantDomain(session), createSession(session));
+
```
Comment (mevan-karu) [inline modules/server-endpoints/language-server/src/main/java/org/wso2/carbon/identity/developer/lsp/debug/runtime/DebugSessionManagerImpl.java:93]: Can we remove this additional space here?

File: modules/server-endpoints/language-server/src/main/java/org/wso2/carbon/identity/developer/lsp/debug/runtime/DebugSessionManagerImpl.java [hunk 2]
```diff
@@ -86,13 +88,29 @@ public void destroy() {
...
+        } else {
+            try {
+                JsonDap jsonDap = new JsonDap();
+                jsonDap.init();
+                ProtocolMessage message = new ProtocolMessage(DAPConstants.DEBUG_ERROR_DEBUG_SESSION_ALIVE);
+                String text = jsonDap.encode(message);
+                session.getBasicRemote().sendText(text);
+
```
Comment (mevan-karu) [inline modules/server-endpoints/language-server/src/main/java/org/wso2/carbon/identity/developer/lsp/debug/runtime/DebugSessionManagerImpl.java:101]: Additional space

File: modules/server-endpoints/language-server/src/main/java/org/wso2/carbon/identity/developer/lsp/debug/runtime/DebugSessionManagerImpl.java [hunk 3]
```diff
@@ -86,13 +88,29 @@ public void destroy() {
...
+            } catch (IOException ex) {
+                log.error("Error on Encoding the message", ex);
+
+            }
+
```
Comment (ivantha) [inline modules/server-endpoints/language-server/src/main/java/org/wso2/carbon/identity/developer/lsp/debug/runtime/DebugSessionManagerImpl.java:106]: Let's remove the additional newline here.

File: modules/server-endpoints/language-server/src/main/java/org/wso2/carbon/identity/developer/lsp/debug/runtime/DebugSessionManagerImpl.java [hunk 4]
```diff
@@ -256,4 +269,14 @@ private void initializeListeners() {
...
+    private String getTenantDomain(Session session) {
+
+        List<String> tenantList = session.getRequestParameterMap().get(DAPConstants.JSON_KEY_FOR_TENANT_DOMAIN);
+        if (tenantList != null && !tenantList.isEmpty()) {
+            return tenantList.get(0);
+        }
+        return null;
+    }
+
```
Comment (ivantha) [inline modules/server-endpoints/language-server/src/main/java/org/wso2/carbon/identity/developer/lsp/debug/runtime/DebugSessionManagerImpl.java:281]: I think we can remove the blank line here too.

File: modules/server-endpoints/language-server/src/main/java/org/wso2/carbon/identity/developer/lsp/debug/runtime/DebugSessionManagerImpl.java (hunk 1 above)
Comment (Sathiyakugan) [reply modules/server-endpoints/language-server/src/main/java/org/wso2/carbon/identity/developer/lsp/debug/runtime/DebugSessionManagerImpl.java:93]: Fixed with #12

File: modules/server-endpoints/language-server/src/main/java/org/wso2/carbon/identity/developer/lsp/debug/runtime/DebugSessionManagerImpl.java (hunk 2 above)
Comment (Sathiyakugan) [reply modules/server-endpoints/language-server/src/main/java/org/wso2/carbon/identity/developer/lsp/debug/runtime/DebugSessionManagerImpl.java:101]: Fixed with #12

File: modules/server-endpoints/language-server/src/main/java/org/wso2/carbon/identity/developer/lsp/debug/runtime/DebugSessionManagerImpl.java (hunk 3 above)
Comment (Sathiyakugan) [reply modules/server-endpoints/language-server/src/main/java/org/wso2/carbon/identity/developer/lsp/debug/runtime/DebugSessionManagerImpl.java:106]: Fixed with https://github.com/wso2-extensions/identity-tools-debugger/pull/17

File: modules/server-endpoints/language-server/src/main/java/org/wso2/carbon/identity/developer/lsp/debug/runtime/DebugSessionManagerImpl.java (hunk 4 above)
Comment (Sathiyakugan) [reply modules/server-endpoints/language-server/src/main/java/org/wso2/carbon/identity/developer/lsp/debug/runtime/DebugSessionManagerImpl.java:281]: Fixed with https://github.com/wso2-extensions/identity-tools-debugger/pull/17

**Pull request merged**